cmd/create/admin/cmd.go

```diff
@@ -222,3 +205,53 @@ func generateRandomPassword(length int) (string, error) {
 
 	return string(pw), nil
 }
+
+func FindExistingClusterAdminIDP(cluster *cmv1.Cluster, r *rosa.Runtime) (
+	htpasswdIDP *cmv1.IdentityProvider, userList *cmv1.HTPasswdUserList) {
+
+	// admin user will now be created in a htpasswd IDP named 'cluster-admin'
+	// It should suffice to look for this idp but for back-compatibility ,
+	// i.e for cases where the cluster was created  with previous versions of rosacli
+	// we are going to search  all htpasswd idps
+
+	r.Reporter.Debugf("Loading cluster's identity providers")
+	r.Reporter.Infof("Loading cluster's identity providers")
+	idps, err := r.OCMClient.GetIdentityProviders(cluster.ID())
+	if err != nil {
+		r.Reporter.Errorf("Failed to get identity providers for cluster '%s': %v", r.ClusterKey, err)
```

Comment:
Just curious, why are we using `r.ClusterKey` in the error, but are using `cluster.ID` when fetching idps?

Reply:
that was not intentional , remnants from the old code I guess. Will fix this

Reply:
I think it is because the user might have supplied the name or ID, but to get identity providers we need to supply the ID, but to show the error back to the user we use whatever he supplied.

Reply:
yes, agree. It seems to make sense to use the clusterKey vs clusterID in the error msg.

As clusterID is a long ID string vs clusterKey is the more readable cluster-name